[BUG] [C++] Pistache compiling Pet example

- [x] Have you provided a full/minimal spec to reproduce the issue?

[-] Have you validated the input using an OpenAPI validator (example)?
- [x]  Have you tested with the latest master to confirm the issue still exists?
- [x]  Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?

Using Release 5.3.1 and trying to create a REST server with pistache. It failes to compile using the default PetStore example.

### openapi-generator version
5.3.1
Also exists on 5.4.x and 6.0.X

### Details
I started with just the petstore.yaml since that was the given example and used the following command:

java -jar openapi-generator-cli.jar generate -i https://raw.githubusercontent.com/openapitools/openapi-generator/master/modules/openapi-generator/src/test/resources/3_0/petstore.yaml -g cpp-pistache-server -o /var/tmp/pistache_pet

Then:
cd /var/tmp/pistache_pet
mkdir build
cd build
cmake -G Ninja ..
ninja

This fails as below. I am surprise by this because I am using the example petstore.yaml which I would have expected is tested against often. So perhaps I am doing a step wrong?

I get the following error:
FAILED: CMakeFiles/api-server.dir/impl/UserApiImpl.cpp.o 
/usr/bin/c++  -I/var/tmp/pistache_pet/external/include -I/var/tmp/pistache_pet/model -I/var/tmp/pistache_pet/api -I/var/tmp/pistache_pet/impl -std=c++17 -pg -g3 -MD -MT CMakeFiles/api-server.dir/impl/UserApiImpl.cpp.o -MF CMakeFiles/api-server.dir/impl/UserApiImpl.cpp.o.d -o CMakeFiles/api-server.dir/impl/UserApiImpl.cpp.o -c /var/tmp/pistache_pet/impl/UserApiImpl.cpp
In file included from /var/tmp/pistache_pet/impl/UserApiImpl.h:29,
                 from /var/tmp/pistache_pet/impl/UserApiImpl.cpp:13:
/var/tmp/pistache_pet/api/UserApi.h:88:66: error: ‘User’ was not declared in this scope; did you mean ‘org::openapitools::server::model::User’?
   88 |     virtual void create_users_with_array_input(const std::vector<User> &user, Pistache::Http::ResponseWriter &response) = 0;
      |                                                                  ^~~~
      |                                                                  org::openapitools::server::model::User
In file included from /var/tmp/pistache_pet/api/UserApi.h:29,
                 from /var/tmp/pistache_pet/impl/UserApiImpl.h:29,
                 from /var/tmp/pistache_pet/impl/UserApiImpl.cpp:13:
/var/tmp/pistache_pet/model/User.h:31:8: note: ‘org::openapitools::server::model::User’ declared here
   31 | class  User
      |        ^~~~
In file included from /var/tmp/pistache_pet/impl/UserApiImpl.h:29,
                 from /var/tmp/pistache_pet/impl/UserApiImpl.cpp:13:
/var/tmp/pistache_pet/api/UserApi.h:88:70: error: template argument 1 is invalid
   88 |     virtual void create_users_with_array_input(const std::vector<User> &user, Pistache::Http::ResponseWriter &response) = 0;
      |                                                                      ^
/var/tmp/pistache_pet/api/UserApi.h:88:70: error: template argument 2 is invalid
/var/tmp/pistache_pet/api/UserApi.h:96:65: error: ‘User’ was not declared in this scope; did you mean ‘org::openapitools::server::model::User’?
   96 |     virtual void create_users_with_list_input(const std::vector<User> &user, Pistache::Http::ResponseWriter &response) = 0;
      |                                                                 ^~~~
      |                                                                 org::openapitools::server::model::User
In file included from /var/tmp/pistache_pet/api/UserApi.h:29,
                 from /var/tmp/pistache_pet/impl/UserApiImpl.h:29,
                 from /var/tmp/pistache_pet/impl/UserApiImpl.cpp:13:
/var/tmp/pistache_pet/model/User.h:31:8: note: ‘org::openapitools::server::model::User’ declared here
   31 | class  User
      |        ^~~~
In file included from /var/tmp/pistache_pet/impl/UserApiImpl.h:29,
                 from /var/tmp/pistache_pet/impl/UserApiImpl.cpp:13:
/var/tmp/pistache_pet/api/UserApi.h:96:69: error: template argument 1 is invalid
   96 |     virtual void create_users_with_list_input(const std::vector<User> &user, Pistache::Http::ResponseWriter &response) = 0;
      |                                                                     ^
/var/tmp/pistache_pet/api/UserApi.h:96:69: error: template argument 2 is invalid
[21/27] Building CXX object CMakeFiles/api-server.dir/api/UserApi.cpp.o




I can fix this specific error by specifying the User namespace. However, I then get a second error which I'm not sure how to fix:

FAILED: CMakeFiles/api-server.dir/api/UserApi.cpp.o 
/usr/bin/c++  -I/var/tmp/pistache_pet/external/include -I/var/tmp/pistache_pet/model -I/var/tmp/pistache_pet/api -I/var/tmp/pistache_pet/impl -std=c++17 -pg -g3 -MD -MT CMakeFiles/api-server.dir/api/UserApi.cpp.o -MF CMakeFiles/api-server.dir/api/UserApi.cpp.o.d -o CMakeFiles/api-server.dir/api/UserApi.cpp.o -c /var/tmp/pistache_pet/api/UserApi.cpp
/var/tmp/pistache_pet/api/UserApi.cpp: In member function ‘void org::openapitools::server::api::UserApi::create_users_with_array_input_handler(const Pistache::Rest::Request&, Pistache::Http::ResponseWriter)’:
/var/tmp/pistache_pet/api/UserApi.cpp:109:14: error: ‘class std::vector<org::openapitools::server::model::User>’ has no member named ‘validate’
  109 |         user.validate();
      |              ^~~~~~~~
/var/tmp/pistache_pet/api/UserApi.cpp: In member function ‘void org::openapitools::server::api::UserApi::create_users_with_list_input_handler(const Pistache::Rest::Request&, Pistache::Http::ResponseWriter)’:
/var/tmp/pistache_pet/api/UserApi.cpp:141:14: error: ‘class std::vector<org::openapitools::server::model::User>’ has no member named ‘validate’
  141 |         user.validate();
      |              ^~~~~~~~
[15/16] Building CXX object CMakeFiles/api-server.dir/main-api-server.cpp.o

--- a/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppPistacheServerCodegen.java
+++ b/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppPistacheServerCodegen.java
@@ -378,7 +378,8 @@ public class CppPistacheServerCodegen extends AbstractCppCodegen {
             return toModelName(openAPIType);
         }
 
-        return openAPIType;
+        String namespace = (String)additionalProperties.get("modelNamespace");
+        return namespace + "::" + openAPIType;
     }
 
     @Override

--- a/modules/openapi-generator/src/main/resources/cpp-pistache-server/api-source.mustache
+++ b/modules/openapi-generator/src/main/resources/cpp-pistache-server/api-source.mustache
@@ -93,8 +93,10 @@ void {{classname}}::{{operationIdSnakeCase}}_handler(const Pistache::Rest::Reque
     {{#hasBodyParam}}
     {{#bodyParam}}
     {{^isPrimitiveType}}
-        nlohmann::json::parse(request.body()).get_to({{paramName}});
-        {{paramName}}.validate();
+        nlohmann::json::parse(request.body()).get_to({{paramName}});{{#isArray}}
+        for (const auto& validationParam : {{paramName}})
+             validationParam.validate();{{/isArray}}{{^isArray}}
+        {{paramName}}.validate();{{/isArray}}
     {{/isPrimitiveType}}
     {{#isPrimitiveType}}
         {{paramName}} = request.body();

--- a/samples/server/petstore/cpp-pistache/api/UserApi.cpp
+++ b/samples/server/petstore/cpp-pistache/api/UserApi.cpp
@@ -106,7 +106,8 @@ void UserApi::create_users_with_array_input_handler(const Pistache::Rest::Reques
     
     try {
         nlohmann::json::parse(request.body()).get_to(body);
-        body.validate();
+        for (const auto& validationParam : body)
+             validationParam.validate();
     } catch (std::exception &e) {
         const std::pair<Pistache::Http::Code, std::string> errorInfo = this->handleParsingException(e);
         response.send(errorInfo.first, errorInfo.second);
@@ -138,7 +139,8 @@ void UserApi::create_users_with_list_input_handler(const Pistache::Rest::Request
     
     try {
         nlohmann::json::parse(request.body()).get_to(body);
-        body.validate();
+        for (const auto& validationParam : body)
+             validationParam.validate();
     } catch (std::exception &e) {
         const std::pair<Pistache::Http::Code, std::string> errorInfo = this->handleParsingException(e);
         response.send(errorInfo.first, errorInfo.second);

--- a/samples/server/petstore/cpp-pistache/api/UserApi.h
+++ b/samples/server/petstore/cpp-pistache/api/UserApi.h
@@ -85,7 +85,7 @@ private:
     /// 
     /// </remarks>
     /// <param name="body">List of user object</param>
-    virtual void create_users_with_array_input(const std::vector<User> &body, Pistache::Http::ResponseWriter &response) = 0;
+    virtual void create_users_with_array_input(const std::vector<org::openapitools::server::model::User> &body, Pistache::Http::ResponseWriter &response) = 0;
     /// <summary>
     /// Creates list of users with given input array
     /// </summary>
@@ -93,7 +93,7 @@ private:
     /// 
     /// </remarks>
     /// <param name="body">List of user object</param>
-    virtual void create_users_with_list_input(const std::vector<User> &body, Pistache::Http::ResponseWriter &response) = 0;
+    virtual void create_users_with_list_input(const std::vector<org::openapitools::server::model::User> &body, Pistache::Http::ResponseWriter &response) = 0;
     /// <summary>
     /// Delete user
     /// </summary>

--- a/samples/server/petstore/cpp-pistache/impl/UserApiImpl.cpp
+++ b/samples/server/petstore/cpp-pistache/impl/UserApiImpl.cpp
@@ -27,10 +27,10 @@ UserApiImpl::UserApiImpl(const std::shared_ptr<Pistache::Rest::Router>& rtr)
 void UserApiImpl::create_user(const User &body, Pistache::Http::ResponseWriter &response) {
     response.send(Pistache::Http::Code::Ok, "Do some magic\n");
 }
-void UserApiImpl::create_users_with_array_input(const std::vector<User> &body, Pistache::Http::ResponseWriter &response) {
+void UserApiImpl::create_users_with_array_input(const std::vector<org::openapitools::server::model::User> &body, Pistache::Http::ResponseWriter &response) {
     response.send(Pistache::Http::Code::Ok, "Do some magic\n");
 }
-void UserApiImpl::create_users_with_list_input(const std::vector<User> &body, Pistache::Http::ResponseWriter &response) {
+void UserApiImpl::create_users_with_list_input(const std::vector<org::openapitools::server::model::User> &body, Pistache::Http::ResponseWriter &response) {
     response.send(Pistache::Http::Code::Ok, "Do some magic\n");
 }
 void UserApiImpl::delete_user(const std::string &username, Pistache::Http::ResponseWriter &response) {

--- a/samples/server/petstore/cpp-pistache/impl/UserApiImpl.h
+++ b/samples/server/petstore/cpp-pistache/impl/UserApiImpl.h
@@ -44,8 +44,8 @@ public:
     ~UserApiImpl() override = default;
 
     void create_user(const User &body, Pistache::Http::ResponseWriter &response);
-    void create_users_with_array_input(const std::vector<User> &body, Pistache::Http::ResponseWriter &response);
-    void create_users_with_list_input(const std::vector<User> &body, Pistache::Http::ResponseWriter &response);
+    void create_users_with_array_input(const std::vector<org::openapitools::server::model::User> &body, Pistache::Http::ResponseWriter &response);
+    void create_users_with_list_input(const std::vector<org::openapitools::server::model::User> &body, Pistache::Http::ResponseWriter &response);
     void delete_user(const std::string &username, Pistache::Http::ResponseWriter &response);
     void get_user_by_name(const std::string &username, Pistache::Http::ResponseWriter &response);
     void login_user(const std::optional<std::string> &username, const std::optional<std::string> &password, Pistache::Http::ResponseWriter &response);

--- a/samples/server/petstore/cpp-pistache/model/Pet.cpp
+++ b/samples/server/petstore/cpp-pistache/model/Pet.cpp
@@ -74,14 +74,14 @@ bool Pet::validate(std::stringstream& msg, const std::string& pathPrefix) const
          
     if (tagsIsSet())
     {
-        const std::vector<Tag>& value = m_Tags;
+        const std::vector<org::openapitools::server::model::Tag>& value = m_Tags;
         const std::string currentValuePath = _pathPrefix + ".tags";
                 
         
         { // Recursive validation of array elements
             const std::string oldValuePath = currentValuePath;
             int i = 0;
-            for (const Tag& value : value)
+            for (const org::openapitools::server::model::Tag& value : value)
             { 
                 const std::string currentValuePath = oldValuePath + "[" + std::to_string(i) + "]";
                         
@@ -187,11 +187,11 @@ void Pet::unsetId()
 {
     m_IdIsSet = false;
 }
-Category Pet::getCategory() const
+org::openapitools::server::model::Category Pet::getCategory() const
 {
     return m_Category;
 }
-void Pet::setCategory(Category const& value)
+void Pet::setCategory(org::openapitools::server::model::Category const& value)
 {
     m_Category = value;
     m_CategoryIsSet = true;
@@ -220,11 +220,11 @@ void Pet::setPhotoUrls(std::vector<std::string> const& value)
 {
     m_PhotoUrls = value;
 }
-std::vector<Tag> Pet::getTags() const
+std::vector<org::openapitools::server::model::Tag> Pet::getTags() const
 {
     return m_Tags;
 }
-void Pet::setTags(std::vector<Tag> const& value)
+void Pet::setTags(std::vector<org::openapitools::server::model::Tag> const& value)
 {
     m_Tags = value;
     m_TagsIsSet = true;

--- a/samples/server/petstore/cpp-pistache/model/Pet.h
+++ b/samples/server/petstore/cpp-pistache/model/Pet.h
@@ -71,8 +71,8 @@ public:
     /// <summary>
     /// 
     /// </summary>
-    Category getCategory() const;
-    void setCategory(Category const& value);
+    org::openapitools::server::model::Category getCategory() const;
+    void setCategory(org::openapitools::server::model::Category const& value);
     bool categoryIsSet() const;
     void unsetCategory();
     /// <summary>
@@ -88,8 +88,8 @@ public:
     /// <summary>
     /// 
     /// </summary>
-    std::vector<Tag> getTags() const;
-    void setTags(std::vector<Tag> const& value);
+    std::vector<org::openapitools::server::model::Tag> getTags() const;
+    void setTags(std::vector<org::openapitools::server::model::Tag> const& value);
     bool tagsIsSet() const;
     void unsetTags();
     /// <summary>
@@ -105,13 +105,13 @@ public:
 protected:
     int64_t m_Id;
     bool m_IdIsSet;
-    Category m_Category;
+    org::openapitools::server::model::Category m_Category;
     bool m_CategoryIsSet;
     std::string m_Name;
 
     std::vector<std::string> m_PhotoUrls;
 
-    std::vector<Tag> m_Tags;
+    std::vector<org::openapitools::server::model::Tag> m_Tags;
     bool m_TagsIsSet;
     std::string m_Status;
     bool m_StatusIsSet;

